docs/app/@theme/services/analytics.service.ts

```diff
@@ -31,9 +31,15 @@ export class NgdAnalytics {
   }
 
   trackEvent(eventName: string, eventVal: string = '') {
-    if (this.enabled) {
+    if (this.enabled && this.window.ga) {
       ga('send', 'event', eventName, eventVal);
     }
+
+    if (!this.window.ga) {
```

Comment:
it is mostly fine, but let's separate the `check` logic from the `track` method.
moreover, we also have the `trackPage` method which should also be wrapped into the `check` logic.